src/App.tsx

```diff
@@ -1,26 +1,137 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, { FormEvent, useEffect, useState } from 'react';
+import { TodoHeader } from './components/TodoHeader';
+import { TodoMain } from './components/TodoMain';
+import { TodoFooter } from './components/TodoFooter';
+import { ErrorBox } from './components/ErrorBox';
+import { Filter } from './utils/Filter';
+import { filterTodos } from './utils/todoFilter';
+import { addTodo, deleteTodo, getTodos, USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import { ErrorText } from './utils/ErrorText';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState(Filter.All);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [completedTodos, setCompletedTodos] = useState<Todo[]>([]);
+
+  const prepedTodos = filterTodos(todos, filter);
+  const activeTodosCount = todos.filter(todo => !todo.completed).length;
+  const isEveryActive =
+    todos.every(todo => todo.completed) && todos.length !== 0;
+  const haveCompleted = todos.some(todo => todo.completed);
```

Comment:
And don't keep all variables in one component, keep them where you actually use them